### PR TITLE
Support injecting a list of Spring beans

### LINKED ISSUE #5668: Support injecting a list of Spring beans
Originally brought up by @tpenakov

### PR DESCRIPTION
**Pull request by geoand**
This is now possible due to the existence of
Arc's `@All`

Fixes: #5668

### CODE REVIEW

**Review by mkouba — APPROVED** [review]

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java [hunk 1]
```diff
@@ -384,6 +385,14 @@ Set<AnnotationInstance> getAnnotationsToAdd(
                                 Collections.singletonList((AnnotationValue.createStringValue("value", value)))));
                     }
                 }
+
+                // in Spring List<SomeBean> means that all instances of SomeBean should be injected
+                if (fieldInfo.type().name().equals(DotNames.LIST)) {
```
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java:390]: Fixed

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java [hunk 2]
```diff
@@ -418,6 +427,13 @@ Set<AnnotationInstance> getAnnotationsToAdd(
                         CDI_INJECT_ANNOTATION,
                         target,
                         Collections.emptyList()));
+                // in Spring List<SomeBean> means that all instances of SomeBean should be injected
+                if (methodInfo.returnType().name().equals(DotNames.LIST)) {
```
Comment (mkouba) [inline extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java:431]: `methodInfo.returnType()`? I'm a bit confused now ;-)

File: extensions/spring-di/deployment/src/test/java/io/quarkus/spring/di/deployment/ListOfBeansTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    @org.springframework.stereotype.Service
+    public static class Foo {
+
+        @Autowired
+        List<Service> services;
+
+        final List<Converter> converters;
+
+        Foo(@All List<Converter> converters) {
```
Comment (mkouba) [inline extensions/spring-di/deployment/src/test/java/io/quarkus/spring/di/deployment/ListOfBeansTest.java:43]: So `@All` is still required for constructor injection?

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java (hunk 2 above)
Comment (geoand) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java:431]: Oops, wrong change

Comment (geoand) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java:431]: Actually is it correct. It just look weird in the UI

File: extensions/spring-di/deployment/src/test/java/io/quarkus/spring/di/deployment/ListOfBeansTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    @org.springframework.stereotype.Service
+    public static class Foo {
+
+        @Autowired
+        List<Service> services;
+
+        final List<Converter> converters;
+
+        @Autowired
```
Comment (geoand) [inline extensions/spring-di/deployment/src/test/java/io/quarkus/spring/di/deployment/ListOfBeansTest.java:42]: This should also work if `@Autowired` is missing, but it currently doesn't. 
This is a minor bug that we can fix if someone comes across it

**Pull request merged**